### PR TITLE
Adds a photobooth

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Adds a photobooth machine to the HoP line which can be used to update your security records photo.
It updates the records of the person's name, which means hiding your identity and wearing an ID will let you change other people's photos. If you aren't hiding your identity and are wearing someone else's ID, it won't update at all because it will be unable to find your record.
There's 2 variants of this machine, one at the HoP line that's tied behind Law Office access (so Lawyers have an extra thing they can help out with if no one's available, but maybe it would be better behind Library access since Curators are our "photographer" role?), and the Security one that requires Security access.
The Security one has a special feature that it adds a height chart behind the player.
image

This machine is used by either right-clicking on it or by pressing a button, so the HoP can do it remotely from their line (except on Tram where it's in the library), and after a few delays it'll update your records automatically to how you look like.

Emagging the machine will remove its access restrictions but every time it updates your photos it will spam camera flashing and blind you.

Sprites

Open
image
Closed
image
Security version on the left, has a red tint on it
image

Video demonstration (old sprites):

https://github.com/tgstation/tgstation/assets/53777086/a2e59f08-2d58-4f5b-b081-e137c7606d35

## Why It's Good For The Game

Current security records has no way to change your photo ID, something that was a feature before it was moved to TGUI. The only alternative is to fully delete your record and make a new one with a photo and the same name, but this shares a major issue with the HTML UI: You're immediately sold out by the fact you have a background. No matter how well you try to cover your tracks, any security officer looking at records will see you are the only photo on the manifest that has a background and even a camera in your hand, which makes it impossible to get away with it unless security essentially takes pity on you.
This opens up the ability to fully mask yourself in records, finally.

This is also better even for non antagonists because you can now properly update your photos to match what you look like in the event of an appearance change or even just a job change, which makes it a benefit for HoPs who likes to keep records as up-to-date as possible.

## Changelog

:cl: JohnFulpWillard, Twox
add: Added a new photobooth machine to the HoP's line.
fix: Things checking for access now checks your off hand, too.
/:cl:

### CODE REVIEW

File: code/datums/records/record.dm [hunk 1]
```diff
@@ -188,18 +206,29 @@
...
+ * then make a picture out of it, then finally create a new photo.
+ */
+/datum/record/crew/proc/make_photo(field_name, orientation, add_height_chart)
 	var/icon/picture_image
 	if(!isicon(character_appearance))
 		var/mutable_appearance/appearance = character_appearance
 		appearance.setDir(orientation)
-
+		if(add_height_chart)
+			appearance.underlays += mutable_appearance('icons/obj/machines/photobooth.dmi', "height_chart", alpha = 125)
```
Comment (Ghommie) [inline code/datums/records/record.dm:231]: I think it'd be more efficient to add the height chart post `getFlatIcon`, so it doesn't get skewed by matrix transform or displacement map filters.

Comment (JohnFulpWillard) [reply code/datums/records/record.dm:231]: Icons don't have underlays so I can't just mutable appearance it once it's been turned into one

Comment (Ghommie) [reply code/datums/records/record.dm:231]: Very well, then at least do this:
```suggestion
		if(add_height_chart)
			appearance.underlays += mutable_appearance('icons/obj/machines/photobooth.dmi', "height_chart", alpha = 125, appearance_flags = RESET_TRANSFORM)
```

**Comment by Wallemations** [comment]
Is this something you climb into or stand next to?

**Comment by JohnFulpWillard** [comment]
Preferably it's something to enter, thinking something like this
image

**Comment by Wallemations** [comment]
> Preferably it's something to enter, thinking something like this, with curtains n stuff
> 
> image
> 
> 

In that case you definitely gotta update the icon

**Comment by JohnFulpWillard** [comment]
I dont got an icon to USE man

**Review by MrMelbert — COMMENTED** [review]
image

Fix the floor decals man

File: code/datums/records/record.dm (hunk 1 above)
Comment (Ghommie) [reply code/datums/records/record.dm]: Oh, you know, I should've also suggested the RESET_COLOR and RESET_ALPHA flags too. My bad for bothering you twice.

**Comment by optimumtact** [comment]
mmmm very flavourful

**Comment by Admiral-Gigaton** [comment]
emag functionality?

hacked functionality? (maybe just produces random pictures? or remove access requirements?)

not required, just adds a little more flavor to a flavorful little mechanic

**Comment by Wallemations** [comment]
> I dont got an icon to USE man 

image

File: code/game/machinery/photobooth.dm [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	base_icon_state = "booth"
+	state_open = TRUE
+	circuit = /obj/item/circuitboard/machine/photobooth
+	light_system = OVERLAY_LIGHT_DIRECTIONAL //Used as a flash here.
+	light_range = 6
+	light_color = COLOR_WHITE
+	light_power = FLASH_LIGHT_POWER
+	light_on = FALSE
+	interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND
+	req_one_access = list(ACCESS_LAWYER)
```
Comment (JohnFulpWillard) [inline code/game/machinery/photobooth.dm:21]:
```suggestion
	req_one_access = list(ACCESS_LAWYER, ACCESS_SECURITY)
```

File: code/game/machinery/photobooth.dm [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+///Called by a timer to turn the light off to end the flash effect.
+/obj/machinery/photobooth/proc/flash_end()
+	set_light_on(FALSE)
+
+
+/obj/machinery/button/photobooth
+	name = "photobooth control button"
+	desc = "Operates the photobooth from a distance, allowing people to update their security record photos."
+	device_type = /obj/item/assembly/control/photobooth_control
+	req_one_access = list(ACCESS_SECURITY, ACCESS_HOP)
```
Comment (JohnFulpWillard) [inline code/game/machinery/photobooth.dm:178]:
```suggestion
	req_one_access = list(ACCESS_SECURITY, ACCESS_LAWYER)
```

**Pull request merged**